bitcoin/src/blockdata/script/witness_program.rs

```diff
@@ -37,33 +44,42 @@ impl WitnessProgram {
         P: TryInto<PushBytesBuf>,
         <P as TryInto<PushBytesBuf>>::Error: PushBytesErrorReport,
     {
+        let program =
+            program.try_into().map_err(|error| Error::InvalidLength(error.input_len()))?;
+        WitnessProgram::from_bytes(version, program.as_bytes())
+    }
+
+    /// Creates a new witness program, copying the content from the given byte slice.
+    pub fn from_bytes(version: WitnessVersion, bytes: &[u8]) -> Result<Self, Error> {
         use Error::*;
 
-        let program = program.try_into().map_err(|error| InvalidLength(error.input_len()))?;
-        if program.len() < 2 || program.len() > 40 {
-            return Err(InvalidLength(program.len()));
+        let program_len = bytes.len();
+        if !(MIN_SIZE..=MAX_SIZE).contains(&program_len) {
```

Comment:
In e8a5ead5f21dbb78883fa66fdafe5f944fe8f151:

nit: I find this really hard to read vs the normal way of just writing out the inequalities. I know clippy advises this but I think we have whitelisted that lint across the whole project. (If we haven't, go ahead and add another `allow` line to src/lib.rs.)

Other than that, ACK from me.

Reply:
I wish we could write `program_len.in_range(MIN_SIZE..=MAX_SIZE)`

Reply:
Fixed